sdk/key_vault/src/client.rs

```diff
@@ -1,73 +1,67 @@
 use crate::KeyVaultError;
-use anyhow::Context;
 use anyhow::Result;
 use azure_core::{TokenCredential, TokenResponse};
+use const_format::formatcp;
+use url::Url;
 
 pub(crate) const PUBLIC_ENDPOINT_SUFFIX: &str = "vault.azure.net";
 pub(crate) const API_VERSION: &str = "7.0";
+pub(crate) const API_VERSION_PARAM: &str = formatcp!("api-version={}", API_VERSION);
 
 /// Client for Key Vault operations - getting a secret, listing secrets, etc.
 ///
 /// # Example
 ///
 /// ```no_run
-/// use azure_key_vault::KeyVaultClient;
+/// use azure_key_vault::KeyClient;
 /// use azure_identity::token_credentials::DefaultCredential;
 /// let creds = DefaultCredential::default();
-/// let client = KeyVaultClient::new(&creds, &"test-key-vault");
+/// let client = KeyClient::with_name(&"test-key-vault", &creds).unwrap();
 /// ```
 #[derive(Debug)]
-pub struct KeyVaultClient<'a, T> {
+pub struct KeyClient<'a, T> {
+    pub(crate) vault_url: Url,
+    pub(crate) endpoint: String,
     pub(crate) token_credential: &'a T,
-    pub(crate) keyvault_name: &'a str,
-    pub(crate) endpoint_suffix: String,
-    pub(crate) keyvault_endpoint: String,
     pub(crate) token: Option<TokenResponse>,
 }
 
-impl<'a, T: TokenCredential> KeyVaultClient<'a, T> {
-    /// Creates a new `KeyVaultClient` with an endpoint suffix. Useful for non-public Azure clouds.
-    /// For the default public environment, use `KeyVaultClient::new`.
+impl<'a, T: TokenCredential> KeyClient<'a, T> {
+    /// Creates a new `KeyClient`.
     ///
     /// # Example
     ///
     /// ```no_run
-    /// use azure_key_vault::KeyVaultClient;
+    /// use azure_key_vault::KeyClient;
     /// use azure_identity::token_credentials::DefaultCredential;
     /// let creds = DefaultCredential::default();
-    /// let client = KeyVaultClient::with_endpoint_suffix(&creds, &"test-keyvault", "vault.azure.net".to_owned());
+    /// let client = KeyClient::new("test-key-vault.vault.azure.net", &creds).unwrap();
```

Comment:
I'm assuming the scheme can be passed given you're calling `Url::parse()`. If not, that's concerning but could be addressed in a separate PR. Samples for creating it in other languages SDKs' are similar to:

```csharp
new KeyClient(new Uri("https://myvault.vault.azure.net")); // optional args elided
```

Granted, all the other languages take the URI as a string, but still with the scheme.